editor/filesystem_dock.cpp

```diff
@@ -256,10 +263,11 @@ bool FileSystemDock::_create_tree(TreeItem *p_parent, EditorFileSystemDirectory
 		// Build the tree.
 		for (const FileInfo &fi : file_list) {
 			TreeItem *file_item = tree->create_item(subdirectory_item);
+			String file_metadata = lpath.path_join(fi.name);
 			file_item->set_text(0, fi.name);
 			file_item->set_structured_text_bidi_override(0, TextServer::STRUCTURED_TEXT_FILE);
-			file_item->set_icon(0, _get_tree_item_icon(!fi.import_broken, fi.type));
-			String file_metadata = lpath.path_join(fi.name);
+			file_item->set_icon(0, _get_tree_item_icon(!fi.import_broken, file_metadata, fi.type));
+			file_item->set_icon_max_width(0, 16 * EDSCALE);
```

Comment:
I think instead of hard-coding 16 we could use width of an existing icon, but for that it would be better to use some cached one (and there is no theme cache yet). I could leave a TODO comment maybe? (or add theme cache I guess)

Reply:
You should use `get_theme_constant(SNAME("class_icon_size"), SNAME("Editor"))` instead.